### ISSUE TITLE
[infra] consider "Lock Threads" for issue tracker

### ISSUE THREAD

**Issue by sigmundch** [dart-lang/sdk#54546]
I'm not sure if this has been considered. The flutter project uses https://github.com/marketplace/actions/lock-threads on their issue tracker to keep closed bugs locked and encourage users to file new issues if they encounter something, even if it relates to an old issue.

Have there been any discussions about using it? Other alternatives we should consider?

**Comment by sortie** [comment]
Do we have any data or anecdotal evidence about how much reopening issues affect us? Perhaps we can get that info from github's issue tracker?

Personally I'm ambivalent towards this, I see the uses, though it's also pretty annoying being pinged on bugs a month after they've been closed. It's generally just good open source custom to not reopen issues needlessly, but that can change a lot with the amount of inexperienced users. Dart doesn't get as many open source reports and contributions as Flutter so that might mean we don't have the same needs.

I haven't looked into alternatives and I don't recall any EngProd discussions about this topic. If we were to go with an implementation, I would by default prefer whatever solution Flutter is using to standardize our tech. :)

**Comment by sigmundch** [comment]
I don't believe we see many issues being reopened, but I do see occasionally a comment on an issue that's been closed long before, with a new post showing a similar issue. Often the similar issue is a new bug, and not really the same as the original post. I believe a lock on the thread will prevent such comments from being posted.

I don't think we have data, but would be curious to see what it looks like. For example, how often do we see a comment posted on a closed issue > 1 month after the issue was closed?

I agree though, I find the post "this issue was locked" from the flutter repo annoying too. Maybe we there is a way to do it silently?
